synapse/handlers/e2e_keys.py

```diff
@@ -977,12 +985,66 @@ def _get_e2e_cross_signing_verify_key(self, user_id, key_type, from_user_id=None
         Raises:
             NotFoundError: if the key is not found
         """
+        user = UserID.from_string(user_id)
         key = yield self.store.get_e2e_cross_signing_key(
             user_id, key_type, from_user_id
         )
+
+        if key is None and self.is_mine(user):
```

Comment:
this needs a whole swathe of comments. What does it mean if we are in this situation, and why is pulling the key directly the correct thing to do?

Reply:
Have added comments, let me know what you think of them.

Reply:
looks great. thanks.